src/stores/api/formatters/ethereum.go

```diff
@@ -60,11 +62,55 @@ func FormatSignTypedDataRequest(request *types.SignTypedDataRequest) *signer.Typ
 	return typedData
 }
 
-func FormatTransaction(tx *types.SignETHTransactionRequest) *ethtypes.Transaction {
-	if tx.To == nil {
-		return ethtypes.NewContractCreation(uint64(tx.Nonce), tx.Value.ToInt(), uint64(tx.GasLimit), tx.GasPrice.ToInt(), tx.Data)
+func FormatTransaction(tx *types.SignETHTransactionRequest) (*ethtypes.Transaction, error) {
+	var txData ethtypes.TxData
+
+	switch tx.TransactionType {
+	case types.LegacyTxType:
+		txData = &ethtypes.LegacyTx{
+			Nonce:    uint64(tx.Nonce),
+			GasPrice: tx.GasPrice.ToInt(),
+			Gas:      uint64(tx.GasLimit),
+			To:       tx.To,
+			Value:    tx.Value.ToInt(),
+			Data:     tx.Data,
+		}
+	case types.AccessListTxType:
+		txData = &ethtypes.AccessListTx{
+			ChainID:    tx.ChainID.ToInt(),
+			Nonce:      uint64(tx.Nonce),
+			GasPrice:   tx.GasPrice.ToInt(),
+			Gas:        uint64(tx.GasLimit),
+			To:         tx.To,
+			Value:      tx.Value.ToInt(),
+			Data:       tx.Data,
+			AccessList: tx.AccessList,
+		}
+	case "", types.DynamicFeeTxType:
```

Comment:
MINOR: I would rather not allow empty values to reach this switch and set a default value on the API service method

Reply:
it's better to use the empty string in the switch IMO since the input param of this function is the request type, in which the value can be empty. This centralizes the decision that the default value is Dynamic fee also in the tests.